openpgp/read.go

```diff
@@ -246,10 +247,14 @@ FindLiteralData:
 				return nil, err
 			}
 		case *packet.OnePassSignature:
-			if !p.IsLast {
+			if prevLast {
 				return nil, errors.UnsupportedError("nested signatures")
```

Comment:
```suggestion
				return nil, errors.UnsupportedError("nested signature packets")
```
(just so that we can distinguish the cause of the error in the future)